collections_hierarchy/mixins.py

```diff
@@ -60,29 +87,36 @@ def __init__(self, initial=None):
 
 class OperableMixin(object):
     def __add__(self, other):
-        pass
+        new_list = deepcopy(self)
+        new_list.data += other.data
+        return new_list
 
     def __iadd__(self, other):
-        pass
+        return self.__add__(other)
```

Comment:
Calling add will cause a deepcopy to be created which is not entirely necessary of __iadd__. Could instead just do:
```python
return self.data += other.data
``` 
but, like __add__ the above does not check that the data is actually stored in the 'data' field of self and other. Need to use ```getattr()```